### ISSUE TITLE
How to resume hyperparameter tuning with Ray Tune?

### ISSUE THREAD

**Issue by EthanV431** [ultralytics/ultralytics#20033]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I was tuning YOLOv11l on a fraction of a dataset of ~40,000 images for object detection when it got interrupted. I have the save parameter set to True, but I'm not sure how to resume the tuning process from where I left off. I can't find any directories labelled ray_results or something like that. All of the tuning runs have been saved in folders labelled tune1, tune2, etc. What do I do?

Code:
`from ultralytics import YOLO

# Load your model (pretrained weights or config)
model = YOLO("yolo11l.pt")

results = model.tune(
    data="/home/ec2-user/data.yaml",
    epochs=30,                # Reduce epochs per trial
    iterations=300,           # Fewer hyperparameter combinations
    optimizer="AdamW",
    grace_period=5,
    plots=True,
    save=True,
    val=True,
    use_ray=True,
    gpu_per_trial=1,
    fraction=0.1,
    cache=True,
    resume=True
)`

**Comment by UltralyticsAssistant** [comment]
👋 Hello @EthanV431, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find comprehensive resources for [hyperparameter tuning](https://docs.ultralytics.com/guides/hyperparameter-tuning/), including examples for using Ray Tune.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug the issue. This includes a small dataset or code snippet that replicates the problem you are encountering.

For your ❓ Question, it seems you're trying to resume a tuning process. Please confirm if the `tune1`, `tune2`, etc., directories contain checkpoints or logs from Ray Tune. Providing more details, like the directory structure and contents, would help us better understand your setup. Additionally, make sure you have correctly set up the `resume=True` argument and verify any Ray Tune-specific configurations in your environment.

Join the Ultralytics community where it suits you best:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧.
- Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/).
- Dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge.

## Upgrade

To ensure you're using the latest features and fixes, upgrade to the latest `ultralytics` package, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in the following verified environments with dependencies preinstalled:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to assist you quickly 😊. An Ultralytics engineer will review your question and provide additional guidance soon.

**Comment by glenn-jocher** [comment]
@EthanV431 looks like today is your lucky, @Y-T-G has a PR open introducing resume functionality for RayTune. This should be published in a new release soon. Please test this and see if it suits your needs!

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in [PR #20037](https://github.com/ultralytics/ultralytics/pull/20037). This update introduces exciting new functionality and improvements, including the ability to resume interrupted hyperparameter tuning sessions with the new `resume=True` option in `model.tune()`. This works seamlessly for both standard and Ray Tune workflows, helping save time and resources by continuing from the last interrupted session instead of starting over. The PR also includes updated documentation and internal logic enhancements to support this feature. 

Credit goes to @Y-T-G for the implementation and @glenn-jocher for contributions to this update. The version has been bumped to `8.3.103`, so you can test these changes by updating to the latest version with:

```
pip install -U ultralytics>=8.3.103
```

Alternatively, you can install directly from the main branch:

```
pip install git+https://github.com/ultralytics/ultralytics.git@main
```

Feel free to test and let us know if this resolves the issue for you. If the issue persists or you encounter any other challenges, additional feedback is always welcome. Thank you 🙏 for bringing this to our attention and helping improve Ultralytics!
